scripts/update_e2e_coverage.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env python3
+
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Updates coverage of TensorFlow e2e tests on all backends.
+
+Example usage: python3 update_e2e_coverage.py build-docs
+"""
+
+import argparse
+import collections
+import os
+import subprocess
+
+
+REFERENCE_BACKEND = 'tf'
+# Assumes that tests are expanded for the tf_also, iree_vmla, iree_llvmjit and
+# iree_vulkan backends.
+BACKENDS_TO_TITLES = collections.OrderedDict([
+    ('tf_also', 'tensorflow'),
+    ('iree_vmla', 'vmla'),
+    ('iree_llvmjit', 'llvm-ir'),
+    ('iree_vulkan', 'vulkan-spirv'),
+])
+
+TEST_SUITES_TO_HEADERS = {
+    '//integrations/tensorflow/e2e:e2e_tests':
+        'End to end TensorFlow tests',
+    '//integrations/tensorflow/e2e/keras:keras_tests':
+        'End to end tests written using tf.keras',
+    '//integrations/tensorflow/e2e/keras:vision_external_tests':
+        'End to end tests of tf.keras.applications vision models',
+}
+
+# Some test suites are generated from a single source. This allows us to point
+# to the right test file when generating test URLs.
+SINGLE_SOURCE_SUITES = {
+    '//integrations/tensorflow/e2e/keras:vision_external_tests':
+        'vision_model_test',
+}
+
+# The symbols to show in the table if the operation is supported or not.
+SUCCESS_ELEMENT = '<span class="success-table-element">✓</span>'
+FAILURE_ELEMENT = '<span class="failure-table-element">✗</span>'
+
+MAIN_URL = 'https://github.com/google/iree/tree/main'
```

Comment:
Ah.. more places to update the branch name... 
I'm conflicted on this one. Copybara transforms currently look for that specific URL pattern so it'll be a little annoying to update.

Up to you if you want to use `main` or `master` here. Now that we're aware of it, it's just another small thing to fix up.

Reply:
Not sure I follow – do you mean that copybara won't update this because it's looking for `https://github.com/google/iree/tree/main/` (extra `/`)?

Reply:
Ah true, there is a slash difference... This one should be fine then.

The general point is to keep with convention and let the tooling (or a single commit) manage the branch renaming in one step. We have some code that rewrites URLs and paths and that code will get upset if changes are made piecemeal (reversibility check will fail without manual edits). That code is generally brittle, so we try to not push it too much :/